suite/tests/client-interface/avx512lazy.dll.c

```diff
@@ -81,6 +83,18 @@ bb_event(void *drcontext, void *tag, instrlist_t *bb, bool for_trace, bool trans
                 prev_was_mov_const = true;
         } else
             prev_was_mov_const = false;
+        /* If libc or ld.so has AVX-512 code we have to relax a check above. */
+        for (int i = 0; i < instr_num_dsts(instr); ++i) {
+            opnd_t dst = instr_get_dst(instr, i);
+            if (opnd_is_reg(dst)) {
+                if (reg_is_strictly_zmm(opnd_get_reg(dst)) ||
+                    reg_is_opmask(opnd_get_reg(dst))
+                        IF_X64(|| reg_is_avx512_extended(opnd_get_reg(dst)))) {
+                    lib_avx512 = true;
```

Comment:
The identifier `lib_avx512` and the comment at L86 makes it seem like this logic detects avx512 usage in libs specifically. But that's not true, right? `lib_avx512` would be set for any avx512 instruction, even one in the app-proper. It's only the usage of `lib_avx512` (used at before_marker) that makes it specific to non-app-proper code.

Maybe s/lib_avx512/seen_avx512/ with a comment at L77 that if it is found to be set, it must be in a lib, since the app hasn't done any avx512 by that point?

Reply:
(minor readability nit)